design/additional-data.feature

```diff
@@ -0,0 +1,40 @@
+Feature: Additional fields synchronization
+  - As a store owner
+  - I want to get subscribers details from store
+  - So I would be able to send personalized emails and filter subscribers
+
+  Entities:
+    Newsletter Subscriber - Person who exists in newsletter subscribers list in Magento. Newsletter
+      subscriber can have subscribed and unsubscribed status.
+    Customer - A person who has created an account in store.
+
+  Rule: Additional information moves from Magento to Smaily
```

Comment:
I have changed it now to selected field and its output relation based logic https://github.com/sendsmaily/Sendsmaily-Sync-for-Magento/pull/8/commits/a8340e586fcf316f2aa8f030cacc7bebec4bb57f 

Few things I still consider changing/improving:
- Field `subscription_status` is always synchronized currently and I would remove it from current logic as it is only required for providing `is_unsubscribed` status. As scenario 1 suggests when no fields are selected there should be no fields added.
- Should the logic between customers/guests be further separated? As the last scenario should be:
> Store owner has selected fields and the Newsletter Subscriber is not a Customer.
 ~~Store owner has selected field(s) that does not have value set in Magento~~
- The default requested values(website, group, first name, last name, email, customer ID) for customers to have are set by Magento default behavior.

```   
# Fields required by Magento are always set.
    Scenario: Store owner has selected field(s) and Customer does not have value set in Magento
      Given Store owner has selected <field>
      And Customer does not have value for the <field>
      When Newsletter Subscribers are synchronized
      Then Customer <field> is synchronized as <value>

      Examples:
        |       field         |       value       |
        | Subscription Type   |       customer    |
        | Customer Group      |       General     |
        | Customer ID         |        138        |
        | Prefix              |        ''         |
        | First Name          |       John        |
        | Last Name           |       Doe         |
        | Gender              |        ''         |
        | Date of Birth       |        ''         |
        | Website             |    Main Website   |
        | Store Name          |     English       |

```

Reply:
We discussed this offline and came up with following solutions for the questions:

> Field subscription_status is always synchronized currently and I would remove it from current logic as it is only required for providing is_unsubscribed status. As scenario 1 suggests when no fields are selected there should be no fields added.

You are right. This field is redundant and it can be removed.

> Should the logic between customers/guests be further separated? The default requested values (website, group, first name, last name, email, customer ID) for customers to have are set by Magento default behavior.

Desided to describe in `value` of scenario examples in greater detail:
```
Examples:
| field             | value                          |
| Subscription Type | Constant value of "customer"   |
| Customer Group    | Name of Magento customer group |
```

Reply:
Changed from example values to description of values https://github.com/sendsmaily/Sendsmaily-Sync-for-Magento/pull/8/commits/f713ba2ae3ce35163edfdec072fa0308524d77fc